source/sigv4.c

```diff
@@ -794,6 +899,332 @@ static SigV4Status_t getCredentialScope( SigV4Parameters_t * pSigV4Params,
+
+        pBufLoc = canonicalRequest->pBufCur;
+        buffRemaining = canonicalRequest->bufRemaining;
+
+        keyLen = canonicalRequest->pHeadersLoc[ headerIndex ].key.dataLen;
+        valLen = canonicalRequest->pHeadersLoc[ headerIndex ].value.dataLen;
+        headerKey = canonicalRequest->pHeadersLoc[ headerIndex ].key.pData;
+
+        for( i = 0; i < keyLen; i++ )
+        {
+            if( isTrimmableSpace( headerKey, i, keyLen, trimKeyLen ) )
+            {
+                /* Cannot copy trimmable space into canonical request buffer. */
+            }
+            /* Remaining buffer space should at least accommodate the character to copy and the trailing ":" */
+            else if( buffRemaining <= 1 )
+            {
+                sigV4Status = SigV4InsufficientMemory;
+                break;
+            }
+            else
+            {
+                *pBufLoc = tolower( headerKey[ i ] );
+                pBufLoc++;
+                trimKeyLen++;
+                buffRemaining -= 1;
+            }
+        }
+
+        if( sigV4Status == SigV4Success )
+        {
+            assert( buffRemaining >= 1 );
+            *pBufLoc = ':';
+            pBufLoc++;
+            buffRemaining -= 1;
+        }
+
+        if( sigV4Status == SigV4Success )
+        {
+            value = canonicalRequest->pHeadersLoc[ headerIndex ].value.pData;
+            trimValueLen = 0;
+
+            for( i = 0; i < valLen; i++ )
+            {
+                if( ( isTrimmableSpace( value, i, valLen, trimValueLen ) ) )
+                {
+                    /* Cannot copy trimmable space into canonical request buffer. */
+                }
+                /* Remaining buffer space should at least accommodate the character to copy and the trailing "\n" */
+                else if( ( buffRemaining <= 1 ) )
+                {
+                    sigV4Status = SigV4InsufficientMemory;
+                    break;
+                }
+                else
+                {
+                    *pBufLoc = value[ i ];
+                    pBufLoc++;
+                    trimValueLen++;
+                    buffRemaining -= 1;
+                }
+            }
+
+            if( sigV4Status == SigV4Success )
+            {
+                assert( buffRemaining >= 1 );
+                *pBufLoc = '\n';
+                pBufLoc++;
+
+                /* Calculate the remaining buffer. 1 is added for '\n' character. */
+                buffRemaining -= 1;
+
+                canonicalRequest->pBufCur = pBufLoc;
+                canonicalRequest->bufRemaining = buffRemaining;
+            }
+        }
+
+        return sigV4Status;
+    }
+
+/*-----------------------------------------------------------*/
+
+    static SigV4Status_t appendCanonicalizedHeaders( size_t headerCount,
+                                                     CanonicalContext_t * canonicalRequest )
+    {
+        size_t noOfHeaders = 0;
```

Comment:
`headerIndex` is more suitable of a name for the iteration over the list of header pairs
```suggestion
        size_t headerIndex= 0;
```